enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj

```diff
@@ -655,3 +656,18 @@
                   (is (= {:rows-affected 1}
                          (mt/user-http-request :rasta :post 200 execute-path
                                                {:parameters {"id" 1}}))))))))))))
+
+(deftest upload-csv-test
+  (let [db-id (u/the-id (mt/db))]
+    (testing "Should be blocked without data access"
+      (perms/grant-application-permissions! (perms-group/all-users) :setting)
+      (mt/with-temporary-setting-values [uploads-enabled      true
+                                         uploads-database-id  db-id
+                                         uploads-schema-name  nil
+                                         uploads-table-prefix "uploaded_magic_"]
+        (with-all-users-data-perms {db-id {:details :no}}
+          (is (thrown-with-msg?
+               clojure.lang.ExceptionInfo
+               #"You don't have permissions to do that\."
+               (api.card-test/upload-example-csv! nil false)))))
```

Comment:
I don't totally understand the permissions being set in this test. Why do application permissions matter? And why are you setting DB detail permissions to `no`?

If you want to specifically disable all data access you can do
```
(with-all-users-data-perms {db-id {:data {:native :none :schemas :none}}}
```